src/app/clusters/power-source-server/power-source-server.h

```diff
@@ -28,61 +28,23 @@ namespace chip {
 namespace app {
 namespace Clusters {
 
-struct PowerSourceClusterInfo
-{
-    PowerSourceClusterInfo() : mClusterEndpoint(kInvalidEndpointId) {}
-    explicit PowerSourceClusterInfo(EndpointId powerClusterEndpointId) : mClusterEndpoint(powerClusterEndpointId) {}
-    void Shutdown()
-    {
-        mEndpointList = Span<EndpointId>();
-        mBuf.Free();
-    }
-    CHIP_ERROR SetEndpointList(Span<EndpointId> endpointList)
-    {
-        Shutdown();
-        if (endpointList.size() == 0)
-        {
-            mEndpointList = Span<EndpointId>();
-            return CHIP_NO_ERROR;
-        }
-        mBuf.Calloc(endpointList.size());
-        if (mBuf.Get() == nullptr)
-        {
-            return CHIP_ERROR_NO_MEMORY;
-        }
-        memcpy(mBuf.Get(), endpointList.data(), endpointList.size() * sizeof(EndpointId));
-        mEndpointList = Span<EndpointId>(mBuf.Get(), endpointList.size());
-        return CHIP_NO_ERROR;
-    }
-    EndpointId mClusterEndpoint = kInvalidEndpointId;
-    Platform::ScopedMemoryBuffer<EndpointId> mBuf;
-    Span<EndpointId> mEndpointList;
-};
-
 class PowerSourceServer
 {
 public:
     static PowerSourceServer & Instance();
 
     // Caller does not need to retain the span past the call point as these are copied into an internal storage
     CHIP_ERROR SetEndpointList(EndpointId powerSourceClusterEndpoint, Span<EndpointId> endpointList);
-    // returns nullptr if there's not endpoint list set for this power source cluster endpoint id.
-    const Span<EndpointId> * GetEndpointList(EndpointId powerSourceClusterEndpoint) const;
-    void Shutdown()
+    CHIP_ERROR ClearEndpointList(EndpointId powerSourceClusterEndpoint)
     {
-        for (size_t i = 0; i < kNumSupportedEndpoints; ++i)
-        {
-            mPowerSourceClusterInfo[i].Shutdown();
-        }
+        return SetEndpointList(powerSourceClusterEndpoint, Span<EndpointId>());
     }
-    size_t GetNumSupportedEndpointLists() const { return kNumSupportedEndpoints; };
+    // returns nullptr if there's not endpoint list set for this power source cluster endpoint id.
+    const Span<EndpointId> * GetEndpointList(EndpointId powerSourceClusterEndpoint) const;
```

Comment:
It's odd to return a pointer. Suggest returning the span by value.

Reply:
We may not actually have a list for the requested endpoint, in which case you get back a nullptr.